pairing/bls12381/point.go

```diff
@@ -0,0 +1,403 @@
+	return "bls12-381.G1: " + hex.EncodeToString(b)
+}
+
+func (k *pointG1) Hash(m []byte) kyber.Point {
+	p, _ := bls12381.NewG1().HashToCurve(m, Domain)
+	k.p = p
+	return k
+
+}
+
+func (k *pointG1) IsInCorrectGroup() bool {
+	return bls12381.NewG1().InCorrectSubgroup(k.p)
+}
+
+// Domain comes from the ciphersuite used by the RFC of this name compatible
+// with the paired library > v18
+var Domain = []byte("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_")
+
+// pointG2 is a kyber.Point holding a G2 point on BLS12-381 curve
+type pointG2 struct {
+	p *bls12381.PointG2
+}
+
+func nullpointG2() *pointG2 {
+	var p bls12381.PointG2
+	return newPointG2(&p)
+}
+
+func newPointG2(p *bls12381.PointG2) *pointG2 {
+	return &pointG2{p: p}
+}
+
+func (k *pointG2) Equal(k2 kyber.Point) bool {
+	return bls12381.NewG2().Equal(k.p, k2.(*pointG2).p)
+}
+
+func (k *pointG2) Null() kyber.Point {
+	return newPointG2(bls12381.NewG2().Zero())
+}
+
+func (k *pointG2) Base() kyber.Point {
+	return newPointG2(bls12381.NewG2().One())
+}
+
+func (k *pointG2) Pick(rand cipher.Stream) kyber.Point {
+	var dst, src [32]byte
+	rand.XORKeyStream(dst[:], src[:])
+	return k.Hash(dst[:])
+}
+
+func (k *pointG2) Set(q kyber.Point) kyber.Point {
+	k.p.Set(q.(*pointG2).p)
+	return k
+}
+
+func (k *pointG2) Clone() kyber.Point {
+	var p bls12381.PointG2
+	p.Set(k.p)
+	return newPointG2(&p)
+}
+
+func (k *pointG2) EmbedLen() int {
+	panic("bls12-381: unsupported operation")
+}
+
+func (k *pointG2) Embed(data []byte, rand cipher.Stream) kyber.Point {
+	panic("bls12-381: unsupported operation")
+}
+
+func (k *pointG2) Data() ([]byte, error) {
+	panic("bls12-381: unsupported operation")
+}
+
+func (k *pointG2) Add(a, b kyber.Point) kyber.Point {
+	aa := a.(*pointG2)
+	bb := b.(*pointG2)
+	bls12381.NewG2().Add(k.p, aa.p, bb.p)
+	return k
+}
+
+func (k *pointG2) Sub(a, b kyber.Point) kyber.Point {
+	aa := a.(*pointG2)
+	bb := b.(*pointG2)
+	bls12381.NewG2().Sub(k.p, aa.p, bb.p)
+	return k
+}
+
+func (k *pointG2) Neg(a kyber.Point) kyber.Point {
+	aa := a.(*pointG2)
+	bls12381.NewG2().Neg(k.p, aa.p)
+	return k
+}
+
+func (k *pointG2) Mul(s kyber.Scalar, q kyber.Point) kyber.Point {
+	if q == nil {
+		q = nullpointG2().Base()
+	}
+	bls12381.NewG2().MulScalarBig(
+		k.p,
+		q.(*pointG2).p,
+		&s.(*mod.Int).V,
+	)
+	return k
+}
+
+func (k *pointG2) MarshalBinary() ([]byte, error) {
+	return bls12381.NewG2().ToCompressed(k.p), nil
+}
+
+func (k *pointG2) UnmarshalBinary(buff []byte) error {
+	var err error
+	k.p, err = bls12381.NewG2().FromCompressed(buff)
+	return err
+}
+
+func (k *pointG2) MarshalTo(w io.Writer) (int, error) {
+	buf, err := k.MarshalBinary()
+	if err != nil {
+		return 0, err
+	}
+	return w.Write(buf)
+}
+
+func (k *pointG2) UnmarshalFrom(r io.Reader) (int, error) {
+	buf := make([]byte, k.MarshalSize())
+	n, err := io.ReadFull(r, buf)
+	if err != nil {
+		return n, err
+	}
+	return n, k.UnmarshalBinary(buf)
+}
+
+func (k *pointG2) MarshalSize() int {
+	return 96
```

Comment:
should be in a constant (at the top)

Reply:
Instead of using an additional constant, this can be simply computed as 2 times the `MarshalSize` on G1.